x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcResultSetMetaData.java

```diff
@@ -110,12 +111,12 @@ public String getCatalogName(int column) throws SQLException {
 
     @Override
     public int getColumnType(int column) throws SQLException {
-        return column(column).type.getVendorTypeNumber();
+        return column(column).isArray ? Types.ARRAY : column(column).type.getVendorTypeNumber();
     }
 
     @Override
     public String getColumnTypeName(int column) throws SQLException {
```

Comment:
This needs double checking. The javadocs indicate if it's a user defined type, it should be a fully qualified name - that is, it should return a class type that potentially can be loaded by the app.
I wonder if in this case, we should return the SQL Array interface or just "ARRAY".

Reply:
> that is, it should return a class type that potentially can be loaded by the app.

Wondering if that's correct though: wouldn't you want to use `getColumnClassName()` for that (which indeed returns a FQ class name, ...`.ARRAY`) instead? 

The docs for this method read `Returns: type name used by the database.` (and then indeed `If the column type is a user-defined type, then a fully-qualified type name is returned`).
But my interpretation is that (1) it has to be the DB-used name and (2) that the `fully-qualified` attribute refers here to a hierarchy (which is a common understanding of this concept), which we lack.

Reply:
> getColumnClassName() returned column's base type, which I believe is incorrect. I'll push and update.

I believe it's correct for non-arrays, in case of an array the proper classname is required - `int[] -> [I`.